File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/SanPhamNoiDungColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SanPhamNoiDungRow } from "./SanPhamNoiDungRow";

export interface SanPhamNoiDungColumns {
    Id: Column<SanPhamNoiDungRow>;
    SanPhamTenSp: Column<SanPhamNoiDungRow>;
    TieuDe: Column<SanPhamNoiDungRow>;
    LoaiSection: Column<SanPhamNoiDungRow>;
    NoiDungHtml: Column<SanPhamNoiDungRow>;
    ThuTu: Column<SanPhamNoiDungRow>;
}

export class SanPhamNoiDungColumns extends ColumnsBase<SanPhamNoiDungRow> {
    static readonly columnsKey = 'SanPhamDB.SanPhamNoiDung';
    static readonly Fields = fieldsProxy<SanPhamNoiDungColumns>();
}

File: wheystore/wheystore.Web/Modules/SanPhamDB/SanPhamNoiDung/SanPhamNoiDungColumns.cs
namespace wheystore.SanPhamDB.Columns;

[ColumnsScript("SanPhamDB.SanPhamNoiDung")]
[BasedOnRow(typeof(SanPhamNoiDungRow), CheckNames = true)]
public class SanPhamNoiDungColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string SanPhamTenSp { get; set; }
    [EditLink]
    public string TieuDe { get; set; }
    public int LoaiSection { get; set; }
    public string NoiDungHtml { get; set; }
    public int ThuTu { get; set; }
}